Android/backend/intra/device.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intra
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	"github.com/Jigsaw-Code/Intra/Android/backend/intra/internal/doh"
+	"github.com/Jigsaw-Code/Intra/Android/backend/intra/internal/sni"
+	"github.com/Jigsaw-Code/outline-sdk/network"
+	"github.com/Jigsaw-Code/outline-sdk/network/lwip2transport"
+)
+
+// SocketProtector is a way to make certain sockets or DNS lookups bypassing the VPN connection. This is only needed
+// for devices running Android versions older than Lollipop (21). Once a socket is protected, data sent through it will
+// go directly to the internet, bypassing the VPN. The Android VpnService implements the protect() method.
+type SocketProtector doh.Protector
+
+type IntraDevice struct {
+	t2s       network.IPDevice
```

Comment:
Since you are just extending the IPDevice, you can do this, so you don't need to specify all method wrappers:
```suggestion
type IntraDevice struct {
       network.IPDevice
```